.github/workflows/docker-image.yml

```diff
@@ -34,6 +42,8 @@ jobs:
     steps:
       - name: Check out the repo
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.ref }}
```

Comment:
By providing this input explicitly we can run the workflow from one ref and build the image for another.